app/controllers/rooms_controller.rb

```diff
@@ -24,17 +22,37 @@ def create
 
   def destroy
     room = Room.find(room_params[:id])
-    render json: room
-    room.users.destroy
-    room.user_rooms.destroy
-    room.room_questions.destroy
     room.destroy
+
+    render json: room
   end
 
   private
 
   def room_params
-    params.require(:room).permit(:room_name, :password, :username, :is_active, :id)
+    params
+      .require(:room)
+      .permit(:room_name, :password, :username, :is_active, :id)
   end
 
+  def create_room(user)
+    room = Room.create({
+      "room_name" => room_params[:room_name],
+      "password" => room_params[:password],
+      "host_id" => user.id,
+      "host_name" => user.username
+    })
+    
+    UserRoom.create({"user_id" => user.id, "room_id" => room.id})
+    
+    Question.all.map do |question|
```

Comment:
If you have no need for a variable, don't even create it.

The diff here doesn't make it clear, but the code before created an unused `question` variable.